Source/JavaScriptCore/runtime/StringPrototype.cpp

```diff
@@ -934,30 +934,19 @@ JSC_DEFINE_HOST_FUNCTION(stringProtoFuncSubstring, (JSGlobalObject* globalObject
     int len = jsString->length();
     RELEASE_ASSERT(len >= 0);
 
-    double start = a0.toNumber(globalObject);
+    double start = a0.toIntegerOrInfinity(globalObject);
     RETURN_IF_EXCEPTION(scope, { });
+    start = std::clamp<double>(start, 0.0, len);
     double end;
-    if (!(start >= 0)) // check for negative values or NaN
-        start = 0;
-    else if (start > len)
-        start = len;
     if (a1.isUndefined())
         end = len;
     else {
-        end = a1.toNumber(globalObject);
+        end = a1.toIntegerOrInfinity(globalObject);
         RETURN_IF_EXCEPTION(scope, { });
-        if (!(end >= 0)) // check for negative values or NaN
-            end = 0;
-        else if (end > len)
-            end = len;
-    }
-    if (start > end) {
-        double temp = end;
-        end = start;
-        start = temp;
+        end = std::clamp<double>(end, 0.0, len);
     }
-    unsigned substringStart = static_cast<unsigned>(start);
-    unsigned substringLength = static_cast<unsigned>(end) - substringStart;
+    auto [substringStart, substringEnd] = std::minmax(start, end);
```

Comment:
If we decide not to do the above, then I think this smaller change might also slightly improve performance by cutting down on the number of conversions from `double` to `unsigned`:
```suggestion
    auto [substringStart, substringEnd] = std::minmax<unsigned>(start, end);
```